### PR TITLE
The Martinez drink from cyberpunk 2077

### PR DESCRIPTION
**Pull request by Shadowtheprotogen546**
## About the PR

Could be a fun reference to add to the game, similar to the richard reference to hotline miami

**Media**

Capture
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Floofi
- add: Added 'The Martinez' as a reference to Cyberpunk 2077.  (1 Ice, 1 Lemon juice, 2 vodka, 2 cola)

### CODE REVIEW

**Review by EmoGarbage404 — APPROVED** [review]
cute. i like it.

**Comment by deltanedas** [comment]
wake the fuck up bartender. we have a drink to make

**Comment by Shadowtheprotogen546** [comment]
> wake the fuck up bartender. we have a drink to make

I will add the "Johnny silverhand" and make this the description

**Comment by metalgearsloth** [comment]
This seems like a licensing issue?

**Comment by asperger-sind** [comment]
> This seems like a licensing issue?

Probably. I guess it could be avoided by just renaming it to something like
"Martin Davidez", and the description as
"The Middlewalker legend. Remembered by a drink, forgotten by a drunk."

**Comment by Shadowtheprotogen546** [comment]
> > This seems like a licensing issue?
> 
> I thought you couldn't copyright a name

Recipes, cocktail or otherwise, cannot, as recipes are considered “facts” and a fact cannot be copyright protected

**Comment by Shadowtheprotogen546** [comment]
> > Names are not copyright protected and the sprite would likely fall under fair use as it is a remix of my own creation with enough difference to clearly be its own remix or spinoff
> 
> Names aren't but fictional characters are treated as a component of a copyrighted work. For example you cannot use the name Sherlock Holmes even though it's just a name. Though in this case it's more of a reference which is allowed. CDPR might never find out it was used, they might like it or they might sue. Who knows?

wouldn't they just send a cease and desist in which case we just remove it from the game?

**Comment by metalgearsloth** [comment]
> wouldn't they just send a cease and desist in which case we just remove it from the game?

We're opting to license everything upstream rather than risking any sort of DMCA.

**Review by keronshb — CHANGES_REQUESTED** [review]
Rename this to something that sounds/looks like David Martinez and this should be good to go. We just want to be extra sure about copyright.

File: Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks.yml [hunk 1]
```diff
@@ -2200,3 +2200,19 @@
...
+  id: DrinkDavidMartinez
+  name: "The Martinez" glass
+  description: The edgerunner legend.  Remembered by a drink, Forgotten by a drunk.
+  components:
+    - type: SolutionContainerManager
+      solutions:
+        drink:
+          maxVol: 30
+          reagents:
+            - ReagentId: DavidMartinez
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks.yml:2215]: you changed reagent to TheMartinez so replace this and rename other stuff to follow

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Recipes/Reactions/drinks.yml [hunk 2]
```diff
@@ -574,3 +574,17 @@
...
+    Cola:
+      amount: 2
+    Vodka:
+      amount: 2
+    Ice:
+      amount: 1
+    JuiceLemon:
+      amount: 1
+  products:
+    DavidMartinez: 6
```
Comment (deltanedas) [inline Resources/Prototypes/Recipes/Reactions/drinks.yml:590]: same here...

**Comment by Shadowtheprotogen546** [comment]
Almost done fixing stuff.  Just need to rename a file and test it, then we should be good to go.

File: Resources/Textures/Objects/Consumable/Drinks/david_martinez.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"version": 1, "size": {"x": 32, "y": 32}, "license": "CC-BY-SA-3.0", "copyright": "https://github.com/discordia-space/CEV-Eris/raw/f7aa28fd4b4d0386c3393d829681ebca526f1d2d/icons/obj/drinks.dmi", "states": [{"name": "icon"}]}
```
Comment (deltanedas) [inline Resources/Textures/Objects/Consumable/Drinks/david_martinez.rsi/meta.json:1]: rsi should be renamed to the_martinez.rsi

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks.yml (hunk 1 above)
Comment (Shadowtheprotogen546) [reply Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks.yml]: yea i saw that and changed it.

File: Resources/Textures/Objects/Consumable/Drinks/david_martinez.rsi/meta.json (hunk 3 above)
Comment (Shadowtheprotogen546) [reply Resources/Textures/Objects/Consumable/Drinks/david_martinez.rsi/meta.json:1]: Im not at my home PC rn.  Ill update it when i get home in a few hours.

**Comment by keronshb** [comment]
@Shadowtheprotogen546 Merge master & fix tests/conflicts then I can merge

**Comment by Shadowtheprotogen546** [comment]
> @Shadowtheprotogen546 Merge master & fix tests/conflicts then I can merge

yeah, sorry.  life's been busy.  I should be able to get it done this weekend tho.

File: Resources/Locale/en-US/reagents/meta/consumable/drink/drinks.ftl [hunk 4]
```diff
@@ -76,3 +76,10 @@ reagent-desc-pilk = A sickening mixture of milk and cola.
 reagent-name-posca = posca
 reagent-desc-posca = Poor warriors' drink from a forgotten era.
 
+
+reagent-name-kvass = kvass
+reagent-desc-kvass = A cool refreshing drink with a taste of socialism.
```
Comment (keronshb) [inline Resources/Locale/en-US/reagents/meta/consumable/drink/drinks.ftl:81]: Just need to remove this since it was just recently removed.

**Review by keronshb — CHANGES_REQUESTED** [review]

File: Resources/Locale/en-US/reagents/meta/consumable/drink/drinks.ftl (hunk 4 above)
Comment (Shadowtheprotogen546) [reply Resources/Locale/en-US/reagents/meta/consumable/drink/drinks.ftl:81]: why was it removed out of curiosity?

Comment (keronshb) [reply Resources/Locale/en-US/reagents/meta/consumable/drink/drinks.ftl:81]: I'm not sure myself.

File: Resources/Textures/Objects/Consumable/Drinks/the_martinez.rsi/meta.json [hunk 5]
```diff
@@ -0,0 +1 @@
+{"version": 1, "size": {"x": 32, "y": 32}, "license": "CC-BY-SA-3.0", "copyright": "https://github.com/discordia-space/CEV-Eris/raw/f7aa28fd4b4d0386c3393d829681ebca526f1d2d/icons/obj/drinks.dmi", "states": [{"name": "icon"}]}
```
Comment (keronshb) [inline Resources/Textures/Objects/Consumable/Drinks/the_martinez.rsi/meta.json:1]:
```suggestion
{"version": 1, "size": {"x": 32, "y": 32}, "license": "CC-BY-SA-3.0", "copyright": "https://github.com/discordia-space/CEV-Eris/commit/93dd3fe8aa166a204ae90c520c273e129b574a21", "states": [{"name": "icon"}]}
```

The copyright should link to a commit for the drink sprite and not the dmi file directly.  I changed this for you.

**Review by keronshb — APPROVED** [review]

**Pull request merged**